### PR TITLE
DEMO only: problem log with tensorflow

### PR DESCRIPTION
**Pull request by tylerjereddy**
* this should not be merged, but is an example of a log
that causes a traceback when attempting to generate a
Python summary report with the feature branch that
adds the runtime `HEATMAP`s to that summary report
in https://github.com/darshan-hpc/darshan/pull/715

* you can generate the report successfully with the
latest `pydarshan-devel` using:
`python -m darshan summary treddy_python_id13682221-61449_4-19-82253-18031969888936241172_1.darshan`

* traceback with the PR feature branch is below the fold

<details>

```
Traceback (most recent call last):
  File "/Users/treddy/github_projects/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/python-3.9.5-u6d3ai52ic4te2cyyfljbk2cfmshv2jv/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/treddy/github_projects/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/python-3.9.5-u6d3ai52ic4te2cyyfljbk2cfmshv2jv/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/__main__.py", line 3, in <module>
    main()
  File "/Users/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/__init__.py", line 164, in main
    mod.main(args)
  File "/Users/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py", line 537, in main
    report_data = ReportData(log_path=log_path)
  File "/Users/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py", line 131, in __init__
    self.register_figures()
  File "/Users/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py", line 351, in register_figures
    heatmap_fig = ReportFigure(
  File "/Users/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py", line 69, in __init__
    self.generate_fig()
  File "/Users/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py", line 96, in generate_fig
    fig = self.fig_func(**self.fig_args)
  File "/Users/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/experimental/plots/plot_dxt_heatmap.py", line 423, in plot_heatmap
    jgrid.ax_marg_y.barh(
  File "/Users/treddy/python_venvs/python_395/lib/python3.9/site-packages/matplotlib/axes/_axes.py", line 2513, in barh
    patches = self.bar(x=left, height=height, width=width, bottom=y,
  File "/Users/treddy/python_venvs/python_395/lib/python3.9/site-packages/matplotlib/__init__.py", line 1361, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/Users/treddy/python_venvs/python_395/lib/python3.9/site-packages/matplotlib/axes/_axes.py", line 2304, in bar
    x, height, width, y, linewidth, hatch = np.broadcast_arrays(
  File "<__array_function__ internals>", line 5, in broadcast_arrays
  File "/Users/treddy/python_venvs/python_395/lib/python3.9/site-packages/numpy/lib/stride_tricks.py", line 538, in broadcast_arrays
    shape = _broadcast_shape(*args)
  File "/Users/treddy/python_venvs/python_395/lib/python3.9/site-packages/numpy/lib/stride_tricks.py", line 420, in _broadcast_shape
    b = np.broadcast(*args[:32])
ValueError: shape mismatch: objects cannot be broadcast to a single shape

```

</details>

* in short, there are six cascade-lake Volta nodes (two GPUs per node),
each has 3 mpi ranks, 2 of those ranks per node are used
for machine learning (1 rank per GPU, and the 3rd rank is
for monitoring GPU memory, usage, temperature, and power
draw using the Python bindings to NVML)

* this log file is a bit large even if we did want to add a proper description/README here

### CODE REVIEW

**Comment by tylerjereddy** [comment]
@shanedsnyder @carns maybe one of you can check this log for me--I seem to be finding evidence of an integer overflow (??) on the byte counts for the runtime `HEATMAP POSIX`

```
hmap_df:
       (0.0, 1.6]  (1.6, 3.2]  (3.2, 4.800000000000001]  (4.800000000000001, 6.4]           (6.4, 8.0]  (8.0, 9.600000000000001]  ...  (176.0, 177.60000000000002]  (177.60000000000002, 179.20000000000002]  (179.20000000000002, 180.8]  (180.8, 182.4]  (182.4, 184.0]  (184.0, 185.60000000000002]
rank                                                                                                                             ...                                                                                                                                                                 
0          43908       90929                 189778380                    291737 -9223372036854526901                         0  ...                    106919923                                         0                            0               0               0                            0
1          43908       90929                 189778380                    354480 -9223372036854592689                         0  ...                       193252                                         0                            0               0               0                            0
2      735987995   683634150                 675228427                 701735283            743551658                 770045185  ...                    582135169                                 582135076                    582134967               0               0                            0
3          43908       90929                 189778380                    290238 -9223372036854606079                         0  ...                       207794                                         0                            0               0               0                            0
4          43908     4219348                 185649961      -9223372036854475783 -9223372036854545662                         0  ...                            0                                         0                            0               0               0                            0
```

It looks like the actual crash may be caused by `HEATMAP STDIO`, which may have a very narrow band of activity followed by nothing?

```
mod: HEATMAP STDIO
mod: HEATMAP
*** y: [ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17]
hmap_df:
       (0.0, 1.6]  (1.6, 3.2]  (3.2, 4.800000000000001]  (4.800000000000001, 6.4]  (6.4, 8.0]  (8.0, 9.600000000000001]  ...  (176.0, 177.60000000000002]  (177.60000000000002, 179.20000000000002]  (179.20000000000002, 180.8]  (180.8, 182.4]  (182.4, 184.0]  (184.0, 185.60000000000002]
rank                                                                                                                    ...                                                                                                                                                                 
0            350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
1            350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
3            350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
4            350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
6            350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
7            350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
9            350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
10           350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
12           350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
13           350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
15           350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0
16           350         260                         0                         0           0                         0  ...                            0                                         0                            0               0               0                            0

[12 rows x 116 columns]
*** width:
 rank
0     1611
1     1611
3     1611
4     1611
6     1611
7     1611
9     1611
10    1611
12    1611
13    1611
15    1611
16    1611
dtype: int64
```

**Comment by tylerjereddy** [comment]
Incidentally, every third rank is just NVML monitoring instead of ML work, and those never seem to show the overflow.

**Comment by shanedsnyder** [comment]
I had a look at the log and confirmed bogus counters with negative values in the heatmap module, e.g.:

```
HEATMAP 15      16592106915301738621    HEATMAP_READ_BIN_17     -9223372036854775808    heatmap:POSIX   UNKNOWN UNKNOWN
```

I'm not sure if somehow overflow is being triggered, or if Darshan memory is somehow being corrupted, leading to nonsensical counter values? In any case, I'd be interested in debugging this further obviously -- is the original ML workload anything we can package up into a reproducer so I can better see what's happening at runtime?

As for the PyDarshan error, it seems to imply some data that's being operated on has mismatched shapes, which I think wouldn't necessarily be related to heatmap values being negative? I quickly scanned the STDIO heatmap data and didn't immediately see different sized heatmap bins (and IIRC, our heatmap code triggers an assertion if any process's module heatmap isn't the same size as other processes). Have you found any more clues on what it thinks is "mismatched" there? Would be good to fix that, too, even if it's only triggered by a log with bogus values.

**Comment by tylerjereddy** [comment]
My understanding is that this log is no longer needed because the underlying bugs in both the Python and C layers were resolved in two separate main repo PRs targeting https://github.com/darshan-hpc/darshan/issues/730, while the Python side of the regression testing is handled by the smaller/more reasonable and already-merged https://github.com/darshan-hpc/darshan-logs/pull/37.

The C regression test isn't merged yet, but it will of course generate its own log file on the fly anyway per https://github.com/darshan-hpc/darshan/pull/756
